File: ios/MawaqitMasjid/Sources/Models/AuthToken.swift
// PR-29 — persisted auth token.
import Foundation

public struct AuthToken: Codable, Equatable {
    public let accessToken: String
    public let refreshToken: String?
    public let expiresAt: Date

    public init(accessToken: String, refreshToken: String? = nil, expiresAt: Date) {
        self.accessToken = accessToken
        self.refreshToken = refreshToken
        self.expiresAt = expiresAt
    }

    public var isExpired: Bool { Date() >= expiresAt }
}

File: android/app/src/main/java/com/forge/mawaqitmasjid/models/AuthToken.kt
// PR-29 — persisted auth token.
package com.forge.mawaqitmasjid.models

import kotlinx.serialization.Serializable

@Serializable
data class AuthToken(
    val accessToken: String,
    val refreshToken: String? = null,
    val expiresAtEpochSeconds: Long,
) {
    val isExpired: Boolean
        get() = System.currentTimeMillis() / 1000 >= expiresAtEpochSeconds
}
